### PR TITLE
Fix some printf formats

### PR DESCRIPTION
**Pull request by kamahen**

### CODE REVIEW

File: src/pl-index.c [hunk 1]
```diff
@@ -2046,7 +2046,7 @@ listIndexGenerations(Definition def, gen_t gen)
 	    for(cr=cl->first_clause; cr; cr=cr->next)
 	    { Clause clause = cr->value.clause;
 
-	      Sdprintf("  %p: [%2d] %8u-%10u%s%s\n",
+	      Sdprintf("  %p: [%2d] %8" PRIuMAX "-%10" PRIuMAX "%s%s\n",
```
Comment (JanWielemaker) [inline src/pl-index.c:2049]: Do we have support for PRIuMAX in Sdprintf?    Better yet, use generationName(), which was added when generations became sometimes huge numbers when in a transaction.

**Comment by JanWielemaker** [comment]
All generations should probably be handled using generationName()?

File: src/pl-index.c (hunk 1 above)
Comment (kamahen) [reply src/pl-index.c]: Done. Please check that this is the correct way to call generationName() -- it appears to allocate something on the stack and I want to ensure that this is a valid thing to do here. (also at line 2060 and pl-proc.c:4118)

**Comment by kamahen** [comment]
I have used `generationName()` in `pl-index.c` and `pl-proc.c` ... there might be other places that should use it, but - if so - they didn't trigger a format mismatch.
Please verify that these are valid places to use `generationName()`, which contains a call to `buffer_string(..., BUF_STACK)`.

**Comment by JanWielemaker** [comment]
Thanks.    Merged after a few more edits.

**Pull request closed without merging**